wss_tools/utils/recenter.py

```diff
@@ -0,0 +1,131 @@
+''' Module to recenter images '''
+import os
+from astropy.io import fits
+from astropy.nddata import block_reduce
+import matplotlib.pyplot as plt
+import numpy as np
+from scipy import ndimage
+
+__all__ = ['recenter']
+
+
+def recenter(images, outputdir, doplot=False):
+    ''' Recenter images based on NIRCam XY (464,1412) if offset > 10px
+    Parameters
+    ----------
+    images: List
+        List of input images to analyze
+    outputdir: String
+        Working directory where QUIP will write the files to
+    doplot: boolean
+        Show plots to the user via a popup
+
+    Returns
+    ----------
+    output_images: List
+        Output images that have been read and/or modified
+    '''
+    output_images = []
+    for im_fn in images:
+        # Open fits
+        hdul = fits.open(im_fn)
+        data = hdul[1].data
+
+        # Rebin image with very big pixels to find the approximate location
+        size = 32
+        rebindata = block_reduce(data, block_size=64, func=np.median)
+
+        # Remove background
+        rebindata -= np.median(rebindata)
+
+        margin_left = 1.5
+        margin_right = 2.5
+        margin_top = 2.5
+        margin_bottom = 1.5
+        imsize = 2048
+
+        # Find the center of mass and cut a box around: we should see the whole
+        # PSF with about 1-2 PSF wide margins
+        com = ndimage.center_of_mass(rebindata)
+        subdata = data[int((com[0] - margin_left) * imsize / size):
+                       int((com[0] + margin_right) * imsize / size),
+                       int((com[1] - margin_bottom) * imsize / size):
+                       int((com[1] + margin_top) * imsize / size)]
+
+        # Rebin again
+        rebinsubdata = block_reduce(subdata, block_size=8, func=np.median)
+
+        # Remove background
+        rebinsubdata -= np.median(rebinsubdata)
+
+        # Find the center of mass and cut a box around: should see the inside
+        # of the PSF
+        com1 = ndimage.center_of_mass(rebinsubdata)
+        margin_left2 = 0.5
+        margin_right2 = 1.5
+        margin_bottom2 = 0.5
+        margin_top2 = 1.5
+        subdata2 = \
+            subdata[int((com1[0] - margin_left2) * imsize / size * 4 / size):
+                    int((com1[0] + margin_right2) * imsize / size * 4 / size),
+                    int((com1[1] - margin_bottom2) * imsize / size * 4 / size):
+                    int((com1[1] + margin_top2) * imsize / size * 4 / size)]
```

Comment:
`\` continuation has been discouraged for a long time.
```suggestion
        subdata2 = subdata[int((com1[0] - margin_left2) * imsize / size * 4 / size):
                           int((com1[0] + margin_right2) * imsize / size * 4 / size),
                           int((com1[1] - margin_bottom2) * imsize / size * 4 / size):
                           int((com1[1] + margin_top2) * imsize / size * 4 / size)]
```